### PR TITLE
Support TLS connection to Kafka

### LINKED ISSUE #178: TLS between kafka-pixy and kafka
Is it supported? 
From the look of https://github.com/mailgun/kafka-pixy/blob/v0.17.0/config/config.go I'd guess not (saramaCfg.Net.TLS is not touched, which means it defaults to disabled). 
If not, this issue is to support it.

### PR DESCRIPTION
**Pull request by ahamidi**
This PR adds support for TLS connections to Kafka. It introduces the following new properties to the `Kafka` configuration:

- `TLSEnabled bool` - Enable/Disable TLS
- `CACertFile string` - location of the CA certificate file
- `ClientCertFile string` - location of the client certificate file
- `ClientCertKeyFile string` - location of the client key file
- `InsecureSkipVerify bool` - Disable hostname verification (required if Kafka cluster uses self-signed certs).

Fixes #178

### CODE REVIEW

**Comment by horkhe** [comment]
@ahamidi so far so good :)

**Comment by ahamidi** [comment]
@horkhe I haven't had a chance to test it against a Kafka cluster yet, but hopefully will get a chance to in the next few days.

I noticed that not all configurations options were exposed directly as CLI flags. Was that a conscious decision to simplify the options? Should I trim the Kafka TLS options to the basics/most common?

**Comment by horkhe** [comment]
We only exposed bare minimum of command line parameters for simplest configuration option. Mostly to allow running kafka-pixy on a dev laptop for testing. In production kafka-pixy is supposed to run with only one cmd argument `--config`. Although I see no harm in exposing the TLS related command line arguments, if only to allow quick tests that TLS connections is working. 

Please let me know how your testing goes. Also please squash all the commits with the commit message "Add support for TLS connection to Kafka".

And thanks again.

@thrawn01 do you have anything to add?

**Comment by thrawn01** [comment]
LGTM!

**Comment by ahamidi** [comment]
I was able to test this against Amazon MSK with TLS authentication enabled and it works as expected.

I'm not sure why the test is failing though (fails in `multiplexer_test.go`)

File: config/config.go [hunk 1]
```diff
@@ -87,6 +87,28 @@ type Proxy struct {
 
 		// Version of the Kafka cluster. Supported versions are 0.10.2.1 - 2.0.0
 		Version KafkaVersion
+
+		// Optionally use TLS when connecting to Kafka. This must be enabled
+		// for following options to be used.
+		TLSEnabled bool `yaml:"tls"`
+
+		// The path to the CA certificate (PEM)
+		CACertFile string `yaml:"ca_certificate_path"`
```
Comment (horkhe) [inline config/config.go]: Let's be consistent and call it `file` in the config too. The same applies to the two following parameters. Respectively `testdata/kafka-tls.yaml` needs to be updated.

**Pull request merged**